src/Forms/ElementalAreaField.php

```diff
@@ -208,4 +212,55 @@ public function performReadonlyTransformation()
             ->setName($this->getName())
             ->addExtraClass('elemental-area--read-only');
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setSubmittedValue($value, $data = null)
+    {
+        // Content comes through as a JSON encoded list through a hidden field.
+        return $this->setValue(Convert::json2array($value));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveInto(DataObjectInterface $dataObject)
+    {
+        /** @var BlocksPage $dataObject */
+        parent::saveInto($dataObject);
+
+        $elementData = $this->Value();
+        $idPrefixLength = strlen(sprintf(ElementalAreaController::FORM_NAME_TEMPLATE, ''));
```

Comment:
When I read this I thought "I wonder if I'd prefer to read it as two variables; `$idPrefix` and `$idPrefixLength`", then I ignored myself until I saw the same pattern used on lines 250-251. The difference is that there both of those variables are used, whereas here they aren't. I'll just leave this here as food for thought (not a request necessarily)

Reply:
Yeah. I usually try to limit variable declaration if it's not going to be used unless it's going to make the code super messy. In this case I opted for a variable because it's better to leave this stuff out of the loop.